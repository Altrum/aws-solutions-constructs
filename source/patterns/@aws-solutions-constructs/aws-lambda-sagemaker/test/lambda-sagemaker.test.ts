/**
 *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

// Imports
import { Stack } from "@aws-cdk/core";
import * as lambda from "@aws-cdk/aws-lambda";
import * as sagemaker from "@aws-cdk/aws-sagemaker";
import * as iam from "@aws-cdk/aws-iam";
import { LambdaToSagemaker } from '../lib';
import { SynthUtils } from '@aws-cdk/assert';
import '@aws-cdk/assert/jest';

// --------------------------------------------------------------
// Test deployment with new Lambda function
// --------------------------------------------------------------
test('Test deployment with new Lambda function', () => {
    // Stack
    const stack = new Stack();
    // Helper declaration
    new LambdaToSagemaker(stack, 'lambda-to-sagemaker-stack', {
        lambdaFunctionProps: {
            runtime: lambda.Runtime.NODEJS_10_X,
            handler: 'index.handler',
            code: lambda.Code.fromAsset(`${__dirname}/lambda`),
            environment: {
                LAMBDA_NAME: 'deployed-function'
            }
        }
    });
    // Assertion 1
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    // Assertion 2
    expect(stack).toHaveResourceLike("AWS::Lambda::Function", {
        Environment: {
            Variables: {
                LAMBDA_NAME: 'deployed-function'
            }
        }
    });
    // Assertion 3
    expect(stack).toHaveResource("AWS::KMS::Key", {
        EnableKeyRotation: true
    });
});

// --------------------------------------------------------------
// Test deployment with override
// --------------------------------------------------------------
test('Test deployment with override', () => {
    // Stack
    const stack = new Stack();
    // Helper declaration
    new LambdaToSagemaker(stack, 'lambda-to-sagemaker-stack', {
        lambdaFunctionProps: {
            runtime: lambda.Runtime.NODEJS_10_X,
            handler: 'index.handler',
            code: lambda.Code.fromAsset(`${__dirname}/lambda`),
        },
        deployInsideVpc: true,
        sagemakerNotebookProps: {
            subnetId: 'subnet-deadbeef',
            securityGroupIds: ['sg-deadbeef']
        }
    });
    expect(stack).toHaveResource("AWS::SageMaker::NotebookInstance", {
        DirectInternetAccess: "Disabled",
        SecurityGroupIds: [
          "sg-deadbeef"
        ],
        SubnetId: "subnet-deadbeef"
    });
});

// --------------------------------------------------------------
// Test deployment with existingNotebookObj
// --------------------------------------------------------------
test('Test deployment with existingNotebookObj', () => {
    // Stack
    const stack = new Stack();
    const sagemakerRole = new iam.Role(stack, "SagemakerRole", {
        assumedBy: new iam.ServicePrincipal("sagemaker.amazonaws.com"),
    });
    const _sagemaker = new sagemaker.CfnNotebookInstance(stack, 'MyNotebookInstance', {
        instanceType: 'ml.c4.2xlarge',
        roleArn: sagemakerRole.roleArn
    });

    const construct = new LambdaToSagemaker(stack, 'lambda-to-sagemaker-stack', {
        lambdaFunctionProps: {
            runtime: lambda.Runtime.NODEJS_10_X,
            handler: 'index.handler',
            code: lambda.Code.fromAsset(`${__dirname}/lambda`),
        },
        existingNotebookObj: _sagemaker
    });
    expect(stack).toHaveResource("AWS::SageMaker::NotebookInstance", {
        InstanceType: "ml.c4.2xlarge",
    });
    expect(construct.vpc).toBeUndefined();
    expect(construct.securityGroup).toBeUndefined();
});

// --------------------------------------------------------------
// Test the properties
// --------------------------------------------------------------
test('Test the properties', () => {
    // Stack
    const stack = new Stack();
    // Helper declaration
    const pattern = new LambdaToSagemaker(stack, 'lambda-to-sagemaker-stack', {
        lambdaFunctionProps: {
            runtime: lambda.Runtime.NODEJS_10_X,
            handler: 'index.handler',
            code: lambda.Code.fromAsset(`${__dirname}/lambda`)
        }
    });
    // Assertion 1
    expect(pattern.lambdaFunction !== null);
    // Assertion 2
    expect(pattern.sagemakerNotebook !== null);
    // Assertion 3
    expect(pattern.sagemakerRole !== null);
    // Assertion 4
    expect(pattern.vpc !== null);
    // Assertion 5
    expect(pattern.securityGroup !== null);
});