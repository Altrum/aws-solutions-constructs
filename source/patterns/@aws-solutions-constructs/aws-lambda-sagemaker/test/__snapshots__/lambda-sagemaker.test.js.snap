// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test deployment with new Lambda function 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19bArtifactHash102890AC": Object {
      "Description": "Artifact hash for asset \\"50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19b\\"",
      "Type": "String",
    },
    "AssetParameters50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19bS3Bucket755D82D7": Object {
      "Description": "S3 bucket for asset \\"50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19b\\"",
      "Type": "String",
    },
    "AssetParameters50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19bS3VersionKeyC58BBBDE": Object {
      "Description": "S3 key for asset version \\"50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19b\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "lambdatosagemakerstackEncryptionKeyDC12EC9F": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "EnableKeyRotation": true,
        "KeyPolicy": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion",
                "kms:GenerateDataKey",
                "kms:TagResource",
                "kms:UntagResource",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "lambdatosagemakerstackLambdaFunction1C80F3C5": Object {
      "DependsOn": Array [
        "lambdatosagemakerstackLambdaFunctionServiceRoleDefaultPolicy30BF06A2",
        "lambdatosagemakerstackLambdaFunctionServiceRole1400E87F",
      ],
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W58",
              "reason": "Lambda functions has the required permission to write CloudWatch Logs. It uses custom policy instead of arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole with more tighter permissions.",
            },
          ],
        },
      },
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19bS3Bucket755D82D7",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19bS3VersionKeyC58BBBDE",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters50c19841308501b4ad09b060159161be80516f28a324405b8b67ee9d9f6bf19bS3VersionKeyC58BBBDE",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "LAMBDA_NAME": "deployed-function",
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "lambdatosagemakerstackLambdaFunctionServiceRole1400E87F",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
        "TracingConfig": Object {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "lambdatosagemakerstackLambdaFunctionPolicy5DD3BE2D": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sagemaker:InvokeEndpoint",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":sagemaker:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":endpoint/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lambdatosagemakerstackLambdaFunctionPolicy5DD3BE2D",
        "Roles": Array [
          Object {
            "Ref": "lambdatosagemakerstackLambdaFunctionServiceRole1400E87F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lambdatosagemakerstackLambdaFunctionServiceRole1400E87F": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "logs:CreateLogGroup",
                    "logs:CreateLogStream",
                    "logs:PutLogEvents",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":logs:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":log-group:/aws/lambda/*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "LambdaFunctionServiceRolePolicy",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lambdatosagemakerstackLambdaFunctionServiceRoleDefaultPolicy30BF06A2": Object {
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W12",
              "reason": "Lambda needs the following minimum required permissions to send trace data to X-Ray.",
            },
          ],
        },
      },
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lambdatosagemakerstackLambdaFunctionServiceRoleDefaultPolicy30BF06A2",
        "Roles": Array [
          Object {
            "Ref": "lambdatosagemakerstackLambdaFunctionServiceRole1400E87F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lambdatosagemakerstackSagemakerNotebook804D9A57": Object {
      "Properties": Object {
        "DirectInternetAccess": "Disabled",
        "InstanceType": "ml.t2.medium",
        "KmsKeyId": Object {
          "Ref": "lambdatosagemakerstackEncryptionKeyDC12EC9F",
        },
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "lambdatosagemakerstackSagemakerRoleF07CFB94",
            "Arn",
          ],
        },
        "SecurityGroupIds": Array [
          Object {
            "Fn::GetAtt": Array [
              "lambdatosagemakerstackSecurityGroup53110AF9",
              "GroupId",
            ],
          },
        ],
        "SubnetId": Object {
          "Ref": "lambdatosagemakerstackVpcPrivateSubnet1Subnet96B8DEB0",
        },
      },
      "Type": "AWS::SageMaker::NotebookInstance",
    },
    "lambdatosagemakerstackSagemakerRoleDefaultPolicyD6A62009": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "sagemaker:CreateTrainingJob",
                "sagemaker:DescribeTrainingJob",
                "sagemaker:CreateModel",
                "sagemaker:DescribeModel",
                "sagemaker:DeleteModel",
                "sagemaker:CreateEndpoint",
                "sagemaker:CreateEndpointConfig",
                "sagemaker:DescribeEndpoint",
                "sagemaker:DescribeEndpointConfig",
                "sagemaker:DeleteEndpoint",
                "sagemaker:DeleteEndpointConfig",
                "sagemaker:InvokeEndpoint",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":sagemaker:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:DescribeLogStreams",
                "logs:GetLogEvents",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":logs:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":log-group:/aws/sagemaker/*",
                  ],
                ],
              },
            },
            Object {
              "Action": "iam:GetRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "lambdatosagemakerstackSagemakerRoleF07CFB94",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "iam:PassRole",
              "Condition": Object {
                "StringLike": Object {
                  "iam:PassedToService": "sagemaker.amazonaws.com",
                },
              },
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "lambdatosagemakerstackSagemakerRoleF07CFB94",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lambdatosagemakerstackSagemakerRoleDefaultPolicyD6A62009",
        "Roles": Array [
          Object {
            "Ref": "lambdatosagemakerstackSagemakerRoleF07CFB94",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lambdatosagemakerstackSagemakerRoleF07CFB94": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "sagemaker.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "lambdatosagemakerstackSecurityGroup53110AF9": Object {
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W5",
              "reason": "Allow notebook users to access the Internet from the notebook",
            },
          ],
        },
      },
      "Properties": Object {
        "GroupDescription": "Default/lambda-to-sagemaker-stack/SecurityGroup",
        "SecurityGroupEgress": Array [
          Object {
            "CidrIp": "0.0.0.0/0",
            "Description": "from 0.0.0.0/0:443",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "lambdatosagemakerstackVpcA8F68D19": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "lambdatosagemakerstackVpcFlowLogD22AC2B2": Object {
      "Properties": Object {
        "DeliverLogsPermissionArn": Object {
          "Fn::GetAtt": Array [
            "lambdatosagemakerstackVpcFlowLogIAMRoleFB771551",
            "Arn",
          ],
        },
        "LogDestinationType": "cloud-watch-logs",
        "LogGroupName": Object {
          "Ref": "lambdatosagemakerstackVpcFlowLogLogGroupA21BF4B0",
        },
        "ResourceId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
        "ResourceType": "VPC",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc",
          },
        ],
        "TrafficType": "ALL",
      },
      "Type": "AWS::EC2::FlowLog",
    },
    "lambdatosagemakerstackVpcFlowLogIAMRoleDefaultPolicyE43C80A0": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
                "logs:DescribeLogStreams",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "lambdatosagemakerstackVpcFlowLogLogGroupA21BF4B0",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "iam:PassRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "lambdatosagemakerstackVpcFlowLogIAMRoleFB771551",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lambdatosagemakerstackVpcFlowLogIAMRoleDefaultPolicyE43C80A0",
        "Roles": Array [
          Object {
            "Ref": "lambdatosagemakerstackVpcFlowLogIAMRoleFB771551",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lambdatosagemakerstackVpcFlowLogIAMRoleFB771551": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "vpc-flow-logs.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lambdatosagemakerstackVpcFlowLogLogGroupA21BF4B0": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "RetentionInDays": 731,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Retain",
    },
    "lambdatosagemakerstackVpcIGW12B7EDB5": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "lambdatosagemakerstackVpcPrivateSubnet1DefaultRouteF72255D8": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet1NATGateway2E9BA91C",
        },
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPrivateSubnet1RouteTable408294FB",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "lambdatosagemakerstackVpcPrivateSubnet1RouteTable408294FB": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "lambdatosagemakerstackVpcPrivateSubnet1RouteTableAssociation9812D900": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPrivateSubnet1RouteTable408294FB",
        },
        "SubnetId": Object {
          "Ref": "lambdatosagemakerstackVpcPrivateSubnet1Subnet96B8DEB0",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "lambdatosagemakerstackVpcPrivateSubnet1Subnet96B8DEB0": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "lambdatosagemakerstackVpcPrivateSubnet2DefaultRouteECD37E27": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet2NATGateway3D1CE2B3",
        },
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPrivateSubnet2RouteTableCE0537CA",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "lambdatosagemakerstackVpcPrivateSubnet2RouteTableAssociation0A450B0A": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPrivateSubnet2RouteTableCE0537CA",
        },
        "SubnetId": Object {
          "Ref": "lambdatosagemakerstackVpcPrivateSubnet2SubnetF22718E8",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "lambdatosagemakerstackVpcPrivateSubnet2RouteTableCE0537CA": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "lambdatosagemakerstackVpcPrivateSubnet2SubnetF22718E8": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "lambdatosagemakerstackVpcPublicSubnet1DefaultRoute09D753CD": Object {
      "DependsOn": Array [
        "lambdatosagemakerstackVpcVPCGW631531EC",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "lambdatosagemakerstackVpcIGW12B7EDB5",
        },
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet1RouteTable7F5470F8",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "lambdatosagemakerstackVpcPublicSubnet1EIPD0E2E8FB": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "lambdatosagemakerstackVpcPublicSubnet1NATGateway2E9BA91C": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "lambdatosagemakerstackVpcPublicSubnet1EIPD0E2E8FB",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet1Subnet780F8270",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "lambdatosagemakerstackVpcPublicSubnet1RouteTable7F5470F8": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "lambdatosagemakerstackVpcPublicSubnet1RouteTableAssociationDD69833D": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet1RouteTable7F5470F8",
        },
        "SubnetId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet1Subnet780F8270",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "lambdatosagemakerstackVpcPublicSubnet1Subnet780F8270": Object {
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W33",
              "reason": "Allow Public Subnets to have MapPublicIpOnLaunch set to true",
            },
          ],
        },
      },
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "lambdatosagemakerstackVpcPublicSubnet2DefaultRoute5FB8391A": Object {
      "DependsOn": Array [
        "lambdatosagemakerstackVpcVPCGW631531EC",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "lambdatosagemakerstackVpcIGW12B7EDB5",
        },
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet2RouteTable25C60364",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "lambdatosagemakerstackVpcPublicSubnet2EIP30CE4AE0": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "lambdatosagemakerstackVpcPublicSubnet2NATGateway3D1CE2B3": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "lambdatosagemakerstackVpcPublicSubnet2EIP30CE4AE0",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet2SubnetF32DBDC3",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "lambdatosagemakerstackVpcPublicSubnet2RouteTable25C60364": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "lambdatosagemakerstackVpcPublicSubnet2RouteTableAssociation635880BD": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet2RouteTable25C60364",
        },
        "SubnetId": Object {
          "Ref": "lambdatosagemakerstackVpcPublicSubnet2SubnetF32DBDC3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "lambdatosagemakerstackVpcPublicSubnet2SubnetF32DBDC3": Object {
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W33",
              "reason": "Allow Public Subnets to have MapPublicIpOnLaunch set to true",
            },
          ],
        },
      },
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "Default/lambda-to-sagemaker-stack/Vpc/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "lambdatosagemakerstackVpcVPCGW631531EC": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "lambdatosagemakerstackVpcIGW12B7EDB5",
        },
        "VpcId": Object {
          "Ref": "lambdatosagemakerstackVpcA8F68D19",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
}
`;
